### PR TITLE
improve Base.show of FoldsView (and co) and add some actual tests for it

### PR DESCRIPTION
**Pull request by Evizero**
Here is an example of how it looks now

```julia
julia> A = kfolds((rand(10,2000), rand(2000)), k=10)
10-fold MLDataPattern.FoldsView of 2000 observations:
  data: (10×2000 Array{Float64,2}, 2000-element Array{Float64,1})
  training: 1800 observations/fold
  validation: 200 observations/fold
  obsdim: (last, last)
```

### CODE REVIEW

File: src/MLDataPattern.jl [hunk 1]
```diff
@@ -54,6 +54,12 @@ export
     eachobs,
     eachbatch
 
+simplify_obsdim(::ObsDim.First) = :first
+simplify_obsdim(::ObsDim.Last) = :last
+simplify_obsdim(::ObsDim.Constant{D}) where {D} = D
+simplify_obsdim(::ObsDim.Undefined) = "NA"
```
Comment (oxinabox) [inline src/MLDataPattern.jl]: Why is this one a string? Where as `:first` and `:last` are symbols.

File: src/MLDataPattern.jl [hunk 2]
```diff
@@ -54,6 +54,12 @@ export
     eachobs,
     eachbatch
 
+simplify_obsdim(::ObsDim.First) = :first
```
Comment (oxinabox) [inline src/MLDataPattern.jl]: I don't think simplify is the right word.
`simplify_obsdim` to me says the output should also be an `ObsDim` but a simpler one.

Possibly   `obsdim_string` ?

**Comment by Evizero** [comment]
fair points

**Comment by Evizero** [comment]
As a side note. Tests depend on https://github.com/JuliaLang/METADATA.jl/pull/11296 being merged

**Pull request merged**